book/learning_materials/50_perception_basics/31_skeletonization/31_skeletonization.md

```diff
@@ -0,0 +1,109 @@
+# Skeleton-based Line Detection {#line-detection-skeletonization status=ready}
+
+## Overview
+
+The [segment-based line detection](#line_detection) pipeline available in the [`line_detection`](https://github.com/duckietown/dt-core/tree/master19/packages/line_detector) package provides a nice and compact representation of the different lines in Duckietown for the downstream state estimation. Although it convenient, this representation does not handle turns well, and can suffer from many false positives. To improve these detections, we would like to compress the lines in a representation that preserves as much information as possible.
+
+In order to retain the morphology of the lines, we can extract their **skeletons**. In [](#fig:pipeline-overview), we show the overall pipeline to detect, filter and compress the lines. The output of this pipeline is a set of skeletons that correspond to the different lines.
+
+<figure figure-id="fig:pipeline-overview">
```

Comment:
@bhairavmehta95 Here is a reference to a figure

Reply:
isn't this supposed to look like:

`<figure id="fig:something>`

https://raw.githubusercontent.com/duckietown/docs-learning_materials/de1c6a58f5ee3d11c088cf9579ba5958247c981a/book/learning_materials/21_systems/36_autonomy_architectures/36_autonomy_architectures.md

Reply:
Awesome, that works! Thank you!